main.c

```diff
@@ -777,6 +779,42 @@ resolve_all_solvables (Pool  *pool,
                     if (!queue_contains (&t, *pp))
                       map_clr (pool->considered, *pp);
 
+                  /* For all modular packages, mask bare RPMs with the same name.
+                   * This needs to take into account what considered is currently set to.
+                   */
+                  Id *available_modular_packages = pool_whatprovides_ptr (pool, pool_str2id (pool, MODPKG_PROV, 1));
+                  for (Id *pp = available_modular_packages; *pp; pp++) {
+                      Solvable *modpkg = pool_id2solvable (pool, *pp);
+                      const gchar *name = pool_id2str (pool, modpkg->name);
+
+                      g_auto(Queue) sel;
+                      queue_init (&sel);
+                      selection_make (pool, &sel, name, SELECTION_NAME);
+                      if (!sel.count) {
+                          /* This should never happen, at least one package
+                           * should match (the modular one). */
+                          continue;
+                      }
+                      g_auto(Queue) q;
+                      queue_init (&q);
+                      selection_solvables (pool, &sel, &q);
+
+                      for (int j = 0; j < q.count; j++) {
+                          Id p = q.elements[j];
+                          Solvable *candidate = pool_id2solvable (pool, p);
+                          gboolean is_modular = FALSE;
+                          for (Id *i = available_modular_packages; *i; i++) {
```

Comment:
there should be function `queue_contains` in fus

Reply:
Good point, but this is iterating over array of `Id`s returned by `pool_whatprovides_ptr`, not a `Queue`.